src/directives/recenter.js

```diff
@@ -0,0 +1,38 @@
+goog.provide('ngeo.recenterDirective');
+
+goog.require('ngeo');
+
+
+/**
+ * @ngdoc directive
+ * @ngname ngeoRecenter
+ */
+ngeo.recenterDirective = function() {
+  return {
+    restrict: 'A',
+    link: function($scope, $element, $attrs) {
+      var mapExpr = $attrs['ngeoRecenterMap'];
+      var map = /** @type {ol.Map} */ ($scope.$eval(mapExpr));
+
+      function recenter(element) {
+        var extent = element.attr('extent');
+        if (extent !== undefined) {
+          map.getView().fit(JSON.parse(extent), map.getSize());
```

Comment:
`$scope.$eval` evaluates a property path over the $scope. 
`extent` is a string you want to convert to json. I don't think it's the same usecase, am I wrong ?